client/hotrod-client/src/main/java/org/infinispan/client/hotrod/marshall/BytesOnlyJBossLikeMarshaller.java

```diff
@@ -0,0 +1,175 @@
+package org.infinispan.client.hotrod.marshall;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.io.ByteBufferImpl;
+import org.infinispan.commons.marshall.BufferSizePredictor;
+import org.infinispan.commons.marshall.Marshaller;
+
+/**
+ * Marshaller that only supports byte[] instances written in JBoss marshaller encoding
```

Comment:
I don't know, I normally just look at `RiverMarshaller` and `RiverUnmarshaller`. It can use standard java serialization, but it will prefix that with a byte indicator saying the next object to read is using default java serialization though.

Reply:
> JBoss Marshalling is a framework which wholly replaces Java Serialization with a new API and optional optimized wire format. The standard wire format can also be used as well - giving compatibility with applications using standard serialization, but without the various bugs present therein, and with a potentially significant performance advantage.

So I guess it's possible to write with jboss-marshalling and read with standard serialization, without any prefix, but it's not the default. Please move the constants to the top and add a reference to `RiverMarshaller` and/or `Protocol` in the javadoc.

Reply:
Yeah I don't know how that is configured to only do java serialization, guessing it doesn't use RiverMarshaller though.

Sure I can move them up top.